ibis/backends/sql/datatypes.py

```diff
@@ -1056,8 +1056,11 @@ class ClickHouseType(SqlglotType):
     def from_ibis(cls, dtype: dt.DataType) -> sge.DataType:
         typ = super().from_ibis(dtype)
         # nested types cannot be nullable in clickhouse
-        if dtype.nullable and not (
-            dtype.is_map() or dtype.is_array() or dtype.is_struct()
+        typ.args["nullable"] = False
+        if (
+            dtype.nullable
+            and not (dtype.is_map() or dtype.is_array() or dtype.is_struct())
+            and typ.args["this"] != typecode.NULLABLE
```

Comment:
This line is to ensure that we don't wrap existing `Nullable`s with another nullable.